homeassistant/components/sensor/fail2ban.py

```diff
@@ -45,7 +42,7 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the fail2ban sensor."""
     name = config.get(CONF_NAME)
     jails = config.get(CONF_JAILS)
-    scan_interval = config.get(CONF_SCAN_INTERVAL)
+    scan_interval = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
```

Comment:
This is the wrong fix. Instead of using the internal timer the platform should run the update at the scan interval set by the entity component. It will use the configuration value or the platform default or the project default in that order. Remove the whole timer logic in the data class.

Reply:
Add a throttle to the data class data parsing method if it needs to be limited.